curl-helper.c

```diff
@@ -1136,6 +1136,181 @@ static int cb_SSH_KEYFUNCTION(CURL *easy,
     return res;
 }
 
+/* Same order as in OCaml */
+curl_sslbackend sslBackendMap[] = {
+#if HAVE_DECL_CURLSSLBACKEND_NONE
+  CURLSSLBACKEND_NONE,
+#else
+  -1,
+#endif
+#if HAVE_DECL_CURLSSLBACKEND_OPENSSL
+  CURLSSLBACKEND_OPENSSL,
+#else
+  -1
+#endif
+#if HAVE_DECL_CURLSSLBACKEND_GNUTLS
+  CURLSSLBACKEND_GNUTLS,
+#else
+  -1,
+#endif
+#if HAVE_DECL_CURLSSLBACKEND_NSS
+  CURLSSLBACKEND_NSS,
+#else
+  -1,
+#endif
+#if HAVE_DECL_CURLSSLBACKEND_GSKIT
+  CURLSSLBACKEND_GSKIT,
+#else
+  -1,
+#endif
+#if HAVE_DECL_CURLSSLBACKEND_WOLFSSL
+  CURLSSLBACKEND_WOLFSSL,
+#else
+  -1,
+#endif
+#if HAVE_DECL_CURLSSLBACKEND_SCHANNEL
+  CURLSSLBACKEND_SCHANNEL,
+#else
+  -1,
+#endif
+#if HAVE_DECL_CURLSSLBACKEND_SECURETRANSPORT
+  CURLSSLBACKEND_SECURETRANSPORT,
+#else
+  -1,
+#endif
+#if HAVE_DECL_CURLSSLBACKEND_MBEDTLS
+  CURLSSLBACKEND_MBEDTLS,
+#else
+  -1,
+#endif
+#if HAVE_DECL_CURLSSLBACKEND_MESALINK
+  CURLSSLBACKEND_MESALINK,
+#else
+  -1,
+#endif
+#if HAVE_DECL_CURLSSLBACKEND_BEARSSL
+  CURLSSLBACKEND_BEARSSL,
+#else
+  -1,
+#endif
+};
+
+#if HAVE_DECL_CURLSSLBACKEND_NONE
+typedef struct CURLsslsetMapping CURLsslsetMapping;
+
+struct CURLsslsetMapping
+{
+  char* name;
+  CURLsslset code;
+};
+
+/* Same order as in OCaml */
+CURLsslsetMapping sslsetMap[] = {
+  {"CURLSSLSET_OK", CURLSSLSET_OK},
+  {"CURLSSLSET_UNKNOWN_BACKEND", CURLSSLSET_UNKNOWN_BACKEND},
+  {"CURLSSLSET_TOO_LATE", CURLSSLSET_TOO_LATE},
+  {"CURLSSLSET_NO_BACKENDS", CURLSSLSET_NO_BACKENDS},
+};
+
+static void raiseSslsetError(CURLsslset err)
+{
+  CAMLparam0();
+  CAMLlocal1(exceptionData);
+  const value *exception;
+  const char *errorString = NULL;
+  int i;
+
+  for (i = 0; i < sizeof(sslsetMap) / sizeof(sslsetMap[0]); i ++) {
+    if (sslsetMap[i].code == err) {
+      errorString = sslsetMap[i].name;
+      break;
+    }
+  }
+
+  if (errorString == NULL)
+    caml_invalid_argument("Unexpected CURLsslset value");
+
+  exceptionData = caml_alloc_tuple(2);
+  Store_field(exceptionData, 0, Val_int(i));
+  Store_field(exceptionData, 1, caml_copy_string(errorString));
+
+  exception = caml_named_value("CurlSslSetException");
+  if (exception == NULL) caml_invalid_argument("CurlSslSetException not registered");
```

Comment:
I wanted to say that this should be done at Curl module init time, but I see all other exceptions are done same way, so it is ok (should be improved separately later)